packages/swapper/src/swappers/RelaySwapper/utils/getTrade.ts

```diff
@@ -0,0 +1,361 @@
+        message: `Relay quote with ${swapSteps.length} swap steps not supported (maximum ${MAXIMUM_SUPPORTED_RELAY_STEPS})`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  const swapStepsContainsMultipleItems = swapSteps.some(
+    step => step.items?.length && step.items.length > 1,
+  )
+
+  // It's uncommon but can happen, log it to mixpanel to ensure we will investigate if it happens too much in the future
+  if (swapStepsContainsMultipleItems) {
+    deps.mixPanel?.track(MixPanelEvent.RelayStepMultipleItems)
+
+    return Err(
+      makeSwapErrorRight({
+        message: `Relay quote with step containing multiple items not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  const slippageTolerancePercentageDecimal = (() => {
+    if (_slippageTolerancePercentageDecimal) return _slippageTolerancePercentageDecimal
+    const destinationSlippageTolerancePercentageDecimal = bnOrZero(
+      slippageTolerance.destination.percent,
+    )
+
+    if (destinationSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(destinationSlippageTolerancePercentageDecimal).toFixed()
+    }
+
+    const originSlippageTolerancePercentageDecimal = bnOrZero(slippageTolerance.origin.percent)
+
+    if (originSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(originSlippageTolerancePercentageDecimal).toFixed()
+    }
+  })()
+
+  const protocolAssetId = relayTokenToAssetId(relayToken)
+
+  const maybeProtocolAsset = relayTokenToAsset(relayToken, deps.assetsById)
+
+  if (maybeProtocolAsset.isErr()) {
+    return Err(maybeProtocolAsset.unwrapErr())
+  }
+
+  const protocolAsset = maybeProtocolAsset.unwrap()
+
+  const isCrossChain = sellAsset.chainId !== buyAsset.chainId
+
+  const maybeAppFeesAsset = relayTokenToAsset(quote.fees.app.currency, deps.assetsById)
+
+  if (maybeAppFeesAsset.isErr()) {
+    return Err(maybeAppFeesAsset.unwrapErr())
+  }
+
+  const appFeesAsset = maybeAppFeesAsset.unwrap()
+
+  const appFeesBaseUnit = (() => {
+    // @TODO: we might need to change this logic when solana and BTC are supported
+    const isNativeCurrencyInput =
+      isNativeEvmAsset(sellAsset.assetId) && sellAsset.chainId === appFeesAsset.chainId
+
+    // For cross-chain: always add back app fees
+    // For same-chain: only add back if input is native currency
+    if (isCrossChain || isNativeCurrencyInput) {
+      return quote.fees.app.amount
+    }
+
+    // cross-chain or same-chain with native currency as input are not applicable
+    return '0'
+  })()
+
+  const relayerFeeRelayToken = quote.fees.relayer.currency
+  const maybeRelayerFeesAsset = relayTokenToAsset(relayerFeeRelayToken, deps.assetsById)
+
+  if (maybeRelayerFeesAsset.isErr()) {
+    return Err(maybeRelayerFeesAsset.unwrapErr())
+  }
+
+  const relayerFeesAsset = maybeRelayerFeesAsset.unwrap()
+
+  const relayerFeesBuyAssetBaseUnit = (() => {
```

Comment:
nitpick vernacular

```suggestion
  const relayerFeesBuyAssetCryptoBaseUnit = (() => {
```

Reply:
Changed in [611391f](https://github.com/shapeshift/web/pull/9188/commits/611391fa4c66eaffa9411c4f7c365d0948be1534)